### PR TITLE
Enable user menu name change

### PR DESCRIPTION
**Pull request by InsanityAutomation**
Ensure all languages have ifndef on user menu name and add config option to show it can be defined from config_adv

### CODE REVIEW

**Comment by InsanityAutomation** [comment]
@thinkyhead it looks like travis timed out on one platform. Not sure how to restart it without making another change.

**Comment by p3p** [comment]
You just click the restart button on the right hand side of the stalled platform, Travis sometimes fails to start one of the jobs, apparently it's the price we have to pay for running the tests in parallel.

**Comment by InsanityAutomation** [comment]
@p3p Thanks, I didnt see it there. I just added the github auth linking, so maybe thats needed for it to show up... ill see next time.

**Comment by p3p** [comment]
I didn't think I had any extended permissions on Marlins Travis, but the restart buttons are always there for me, if I do I wish they extended to purging the caches too.

**Comment by InsanityAutomation** [comment]
Might just be that fancy member tag youve got there lol

File: Marlin/src/lcd/ultralcd.cpp [hunk 1]
```diff
@@ -1125,7 +1125,11 @@ void lcd_quick_feedback(const bool clear_buttons) {
     MENU_BACK(MSG_WATCH);
 
     #if ENABLED(CUSTOM_USER_MENUS)
-      MENU_ITEM(submenu, MSG_USER_MENU, _lcd_user_menu);
+      #if ENABLED(CUSTOM_MSG_USER_MENU)
```
Comment (AnHardt) [inline Marlin/src/lcd/ultralcd.cpp]: `#ifdef` not `#if ENABLED()` . `CUSTOM_MSG_USER_MENU` is neither `true` nor `false`.

What is that good for? Is `MSG_USER_MENU` used somewhere else?

**Review by InsanityAutomation — COMMENTED** [review]
It's not used anywhere else, but thinkyhead wanted it a separate variable instead of overloading the one in the language definition.

File: Marlin/src/lcd/ultralcd.cpp (hunk 1 above)
Comment (thinkyhead) [reply Marlin/src/lcd/ultralcd.cpp]: Override of a single menu title without needing to alter every language file, FTW.

File: Marlin/src/lcd/language/language_en.h [hunk 2]
```diff
@@ -190,9 +190,7 @@
 #ifndef MSG_EDITING_STOPPED
   #define MSG_EDITING_STOPPED                 _UxGT("Mesh Editing Stopped")
 #endif
-#ifndef MSG_USER_MENU
-  #define MSG_USER_MENU                       _UxGT("Custom User Commands")
-#endif
+#define MSG_USER_MENU                       _UxGT("Custom User Commands")
```
Comment (thinkyhead) [inline Marlin/src/lcd/language/language_en.h]: This should be left as it was.

File: Marlin/src/lcd/language/language_eu.h [hunk 3]
```diff
@@ -79,7 +79,7 @@
 #define MSG_LEVEL_CORNERS                   _UxGT("Ertzak berdindu")
 #define MSG_NEXT_CORNER                     _UxGT("Hurrengo ertza")
 //#define MSG_EDITING_STOPPED               _UxGT("Mesh Editing Stopped")
-//#define MSG_USER_MENU                     _UxGT("Custom Commands")
+#define MSG_USER_MENU                       _UxGT("Custom Commands")
```
Comment (thinkyhead) [inline Marlin/src/lcd/language/language_eu.h]: Remove this line altogether. If a string has no translation then it should simply be omitted.

Comment (InsanityAutomation) [reply Marlin/src/lcd/language/language_eu.h]: Done

File: Marlin/src/lcd/language/language_en.h (hunk 2 above)
Comment (InsanityAutomation) [reply Marlin/src/lcd/language/language_en.h]: Done

**Comment by AnHardt** [comment]
For me it should look like

Configuration.h
```cpp
...
//#define MSG_CUSTOM_USER_MENU "Bla bla"
...
```

language_en.h
```cpp
...
#ifdef MSG_CUSTOM_USER_MENU
  #undef MSG_USER_MENU
  #define MSG_USER_MENU MSG_CUSTOM_USER_MENU
#else 
 #ifndef MSG_USER_MENU
    #define MSG_USER_MENU "User Menu"
  #endif 
#endif 
...
```

used as:
```cpp
print(MSG_USER_MENU);
```

That should cleanly override MSG_USER_MENU with  MSG_CUSTOM_USER_MENU if that is defined, and if not defined in another language_xx.h use the english version.

**Comment by InsanityAutomation** [comment]
@AnHardt ill leave that decision up to however thinkyhead wants me to put it. This is how we discussed it going on discord last night.

**Comment by thinkyhead** [comment]
@AnHardt — That also works! I doubt there will ever be two custom user menus, but might as well set a precedent for how other strings are overridden.

**Comment by AnHardt** [comment]
@thinkyhead 
I still believe in You as being The macro/preprocessor wizard. :-)

**Comment by thinkyhead** [comment]
Perhaps with Éduardo's help I can also become a template wizard… They can also do wonders.

**Pull request merged**